actioncable/app/assets/javascripts/action_cable/connection.coffee

```diff
@@ -46,12 +51,21 @@ class ActionCable.Connection
     else
       @open()
 
+  getProtocol: ->
+    @webSocket?.protocol
+
+  isProtocolSupported: ->
+    @getProtocol() in protocols and @getProtocol() isnt unsupported_protocol
```

Comment:
Still don't understand why we need to pass `unsupported_protocol` explicitly around in all these places. If `protocols` is a by-priority list of protocols for the client to respond to. Then wouldn't on the last protocol really indicate nothing more to negotiate; close the connection altogether. To me that makes more sense than jumping through hoops with this variant of a protocol.

To top it off, I'd like to add the unsupported protocol back to `ActionCable::INTERNAL[:protocols]`, so we can think of it as a manifest of the negotiation order.